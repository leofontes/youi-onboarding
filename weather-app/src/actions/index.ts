import * as tmdbActions from './tmdbActions';
import * as youtubeActions from './youtubeActions';

export {
  tmdbActions as tmdb,
  youtubeActions as youtube,
};
