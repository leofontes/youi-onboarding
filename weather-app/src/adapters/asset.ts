export interface Asset {
  images: AssetImages;
  thumbs: AssetImages;
  title: string;
  details: string;
  extra: string;
  similar: Asset[];
  key: string;
  id: number | string;
  youtubeId: string;
  type: AssetType;
  genres?: { id: number; name: string }[];
  runtime: number;
  seasons: number;
  episodes: number;
  live?: Live;
  releaseDate: string;
}

interface Live {
  duration: number;
  elapsed: number;
  channel: { name: string; width: number };
  streams: {
    uri: string;
    type: string;
    id: string;
  }[];
}

export interface AssetImages {
  Poster: string;
  Backdrop: string;
}

export enum AssetType {
  TV = 'tv',
  MOVIE = 'movie',
  AD = 'ad',
}
