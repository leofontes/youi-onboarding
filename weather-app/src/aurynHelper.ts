import { DeviceInfo, Ref } from '@youi/react-native-youi';
import { NativeModules, findNodeHandle } from 'react-native';
import { once } from 'lodash';

const systemName = DeviceInfo.getSystemName();
const { Cloud, RefUtils } = NativeModules;

interface AurynHelper {
  hasHardwareBackButton: boolean;
  isRoku: boolean;
  updateCloudScene: (component: React.RefObject<any>) => void;
  togglePointerEvents: (ref: Ref, enabled: boolean) => boolean;
  toggleParentPointerEvents: (ref: Ref, enabled: boolean) => boolean;
}

export const AurynHelper: AurynHelper = {
  hasHardwareBackButton: !['iOS'].includes(systemName),
  isRoku: Cloud.isCloudServer,
  updateCloudScene: (component) => {
    if (!Cloud.isCloudServer || !component.current) return;
    setTimeout(() => {
      Cloud.exportSubtree(findNodeHandle(component.current), false);
      Cloud.sendFocusMap();
    }, 300);
  },
  togglePointerEvents: once((ref: Ref, enabled: boolean) => {
    RefUtils.setPointerEvents(findNodeHandle(ref), enabled);
    return true;
  }),
  toggleParentPointerEvents: once((ref: Ref, enabled: boolean) => {
    RefUtils.setParentCompositionPointerEvents(findNodeHandle(ref), enabled);
    return true;
  }),
};
