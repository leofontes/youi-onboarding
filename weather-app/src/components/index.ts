export { BackButton } from './backButton';
export { DiscoverContainer } from './discoverContainer';
export { List } from './list';
export { ListItem } from './listitem';
export { Timeline } from './timeline';
export { ToggleButton } from './toggleButton';
export { ToggleGroup } from './toggleGroup';
export { NavigationBar } from './navigationBar';
export { TvContainer } from './tvContainer';
export { Error } from './error';
export { withOrientation } from './withOrientation';
export { LiveContainer } from './liveContainer';
export { LiveListItem } from './liveListitem';
export { BackgroundImage } from './backgroundImage';
