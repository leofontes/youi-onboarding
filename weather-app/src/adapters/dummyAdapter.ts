import { Asset, AssetType } from './asset';

const fromApi = (fillSimilar = true): Asset => ({
  images: {
    Poster: 'https://via.placeholder.com/200x300',
    Backdrop: 'https://via.placeholder.com/1068x600',
  },
  thumbs: {
    Poster: 'https://via.placeholder.com/400x600',
    Backdrop: 'https://via.placeholder.com/534x300',
  },
  title: 'You.i TV',
  details:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
  extra: 'Director: Alan Smithee | Starring: John Doe, Jane Doe',
  similar: fillSimilar ? Array(5).fill(fromApi(false)) : [],
  key: 'keystring',
  id: 1234567,
  youtubeId: 'nO_DIwuGBnA',
  type: AssetType.TV,
  genres: [28, 12],
});

export { fromApi };
