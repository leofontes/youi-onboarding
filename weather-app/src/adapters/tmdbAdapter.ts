/* eslint-disable camelcase */
import { Asset, AssetType } from './asset';
import { genres } from '../data/genres';

interface TmdbCredits {
  crew: TmdbCrew[];
  cast: TmdbCrew[];
}

interface TmdbCrew {
  job: string;
  name: string;
}

export interface TmdbResults<T> {
  results: T[];
}

interface TmdbVideo {
  key: string;
}

export interface TmdbApi {
  poster_path: string;
  backdrop_path: string;
  name: string;
  title: string;
  overview: string;
  credits: TmdbCredits;
  id: number;
  videos: TmdbResults<TmdbVideo>;
  similar: TmdbResults<TmdbApi>;
  original_language: string;
  adult: boolean;
  media_type: 'tv' | 'movie';
  type: 'tv' | 'movie';
  genres: { id: number; name: string }[];
  genre_ids: number[];
  runtime: number;
  number_of_seasons: number;
  number_of_episodes: number;
  release_date?: string;
  first_air_date?: string;
}

const getFeaturedText = (credits: TmdbCredits): string => {
  if (!credits) return '';

  const director = credits.crew.find((it: TmdbCrew) => it.job === 'Director');
  const screenplay = credits.crew.find((it: TmdbCrew) => it.job === 'Screenplay');
  const cast = credits.cast
    .slice(0, 2)
    .map((it: TmdbCrew) => it.name)
    .join('\nStarring\n\n');
  if (director && screenplay) return `${director.name}\nDirector\n\n${screenplay.name}\nScreenplay\n\n${cast}\nStarring`;
  else if (director) return `${director.name}\nDirector\n\n${cast}\nStarring`;
  else if (screenplay) return `${screenplay.name}\nScreenplay\n\n${cast}\nStarring`;

  return `${cast}\nStarring`;
};

const fromApi = (item: TmdbApi): Asset => {
  const type = 'name' in item ? AssetType.TV : AssetType.MOVIE;
  const itemGenres = item.genres || item.genre_ids.map(id => genres[type].find(it => it.id === id));
  return ({
    images: {
      Poster: `http://image.tmdb.org/t/p/w1280/${item.poster_path}`,
      Backdrop: `http://image.tmdb.org/t/p/w1280/${item.backdrop_path}`,
    },
    thumbs: {
      Poster: `http://image.tmdb.org/t/p/w500/${item.poster_path}`,
      Backdrop: `http://image.tmdb.org/t/p/w500/${item.backdrop_path}`,
    },
    title: item.name || item.title,
    details: item.overview,
    extra: getFeaturedText(item.credits),
    similar: item.similar && item.similar.results ? item.similar.results.map((it: TmdbApi) => fromApi(it)) : [],
    key: item.id.toString(),
    id: item.id,
    youtubeId: item.videos && item.videos.results.length ? item.videos.results[0].key : 'nO_DIwuGBnA',
    type,
    genres: itemGenres,
    runtime: item.runtime,
    seasons: item.number_of_seasons,
    episodes: item.number_of_episodes,
    releaseDate: item.release_date || item.first_air_date || ''
  });
}
export { fromApi };
