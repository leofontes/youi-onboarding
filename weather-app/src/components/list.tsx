import React from 'react';
import { ListRef, ListItem as ListItemType, FormFactor } from '@youi/react-native-youi';
import { DiscoverContainer, ListItem, TvContainer, LiveContainer } from '.';
import { isEqual, chunk } from 'lodash';
import { ImageType, ListItemPressEvent, ListItemFocusEvent } from './listitem';
import { Asset } from '../adapters/asset';
import { AurynHelper } from '../aurynHelper';
import { View } from 'react-native';

export enum ListType {
  Featured, Poster, Grid, LargeBackdrop, WideBackdrop, SmallBackdrop, Live, Search, None
}

interface ListProps<T>  {
  type: ListType;
  focusable?: boolean;
  onPressItem?: ListItemPressEvent;
  onFocusItem?: ListItemFocusEvent;
  name: string;
  horizontal: boolean;
  data: T[];
  snapToAlignment?: "start" | "center" | "end";
  numColumns?: number;
  extraData?: any;
}

interface ImageSettings extends ImageType { length: number }

export class List extends React.Component<ListProps<Asset>> {
  static defaultProps = {
    extraData: [],
    type: ListType.None,
  };

  listRef = React.createRef<ListRef<Asset>>();

  chunkedListRef = React.createRef<ListRef<Asset[]>>();

  getImageSettings = (): ImageSettings => {
    switch (this.props.type) {
      case ListType.Poster:
        return { type: 'Poster', size: 'Small', length: 400 };
      case ListType.Grid:
        return { type: 'Backdrop', size: 'Small', length: 534 };
      case ListType.LargeBackdrop:
      case ListType.Featured:
        return { type: 'Backdrop', size: 'Large', length: 1068 };
      case ListType.Search:
        return { type: 'Poster', size: 'Basic', length: 366 };
      case ListType.WideBackdrop:
        return { type: 'Backdrop', size: 'Wide', length: 606};
      default:
        return { type: 'Backdrop', size: 'Small', length: 534 };
    }
  };

  scrollToIndex = (index: number) => {
    this.listRef.current?.scrollToIndex({index, animated: !AurynHelper.isRoku});
    this.chunkedListRef.current?.scrollToIndex({index, animated: !AurynHelper.isRoku});
  }

  imageSettings: ImageSettings = this.getImageSettings();

  chunkSize: number = this.props.type === ListType.Featured ? 3 : 2;

  shouldComponentUpdate(nextProps: ListProps<Asset>) {
    if (AurynHelper.isRoku) return true;

    if (!isEqual(nextProps.extraData, this.props.extraData)) return true;

    return nextProps.focusable !== this.props.focusable;
  }

  componentDidUpdate(prevProps: ListProps<Asset>) {
    if (!this.props.focusable || this.props.extraData !== prevProps.extraData) {
      AurynHelper.updateCloudScene(this.listRef);
      AurynHelper.updateCloudScene(this.chunkedListRef);
    }
  }

  getItemLayout = (_: object, index: number) => ({
    length: this.imageSettings.length,
    offset: this.imageSettings.length * index,
    index,
  });

  keyExtractor = (_item: Asset[], index: number) => index.toString();

  renderItem = ({ item }: ListItemType<Asset>) => (
    <ListItem
      focusable={this.props.focusable}
      onFocus={this.props.onFocusItem}
      onPress={this.props.onPressItem}
      data={item}
      imageType={this.getImageSettings()}
    />
  );

  renderHeaderItem = () => {
    if(FormFactor.isHandset) {
      switch (this.props.type) {
        case ListType.Grid:
        case ListType.Live:
          return (<View style={{height:232}}></View>);
        default:
          return null;
      }
    } else {return null;}
  }

  isHorizontal = () => {
    if (this.props.numColumns)
      return false;
    else
      return !FormFactor.isHandset || this.props.type === ListType.SmallBackdrop
  }

  renderMultipleItems = ({ item, index }: ListItemType<Asset[]>) => {
    if (this.props.type === ListType.Featured) {
      return (
        <DiscoverContainer
          focusable={this.props.focusable}
          onPressItem={this.props.onPressItem}
          onFocusItem={this.props.onFocusItem}
          data={item}
          index={index}
        />
      );
    }

    if (this.props.type === ListType.Grid) {
      return (
        <TvContainer
          focusable={this.props.focusable}
          onPressItem={this.props.onPressItem}
          onFocusItem={this.props.onFocusItem}
          data={item}
        />
      );
    }

    if (this.props.type === ListType.Live) {
      return (
        <LiveContainer
          onPress={this.props.onPressItem}
          onFocus={this.props.onFocusItem}
          focusable={this.props.focusable}
          data={item}
        />
      );
    }

    return null;
  };

  render() {
    const { type, name } = this.props;
    const data = this.props.data.filter(it => it.details !== "" || it.images.Backdrop.indexOf('null') === -1);

    if ([ListType.Featured, ListType.Grid, ListType.Live].includes(type)) {
      return (
        <ListRef
          name={name}
          data={chunk(data, this.chunkSize)}
          ref={this.chunkedListRef}
          horizontal={!FormFactor.isHandset && type !== ListType.Live}
          initialNumToRender={AurynHelper.isRoku ? 100 : 2}
          getItemLayout={this.getItemLayout}
          renderItem={this.renderMultipleItems}
          // ListHeaderComponent={this.renderHeaderItem}
          extraData={this.props.focusable}
          keyExtractor={this.keyExtractor}
          snapToAlignment={this.props.snapToAlignment}
        />
      );
    }

    return (
      <ListRef
        name={name}
        data={data}
        ref={this.listRef}
        horizontal={this.isHorizontal()}
        initialNumToRender={AurynHelper.isRoku ? 100 : 2}
        getItemLayout={this.getItemLayout}
        renderItem={this.renderItem}
        ListHeaderComponent={this.renderHeaderItem}
        extraData={this.props.focusable}
        snapToAlignment={this.props.snapToAlignment}
        numColumns={this.props.numColumns}
      />
    );
  }
}
