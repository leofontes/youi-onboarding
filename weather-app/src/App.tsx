import React from 'react';
import store from './store';
import Stack from './navigation';
import { FormFactor } from '@youi/react-native-youi';
import { Provider } from 'react-redux';
import { BackgroundImage } from './components';

export default class YiReactApp extends React.Component {
  render = () =>
    <Provider store={store}>
      <BackgroundImage source={!FormFactor.isHandset ? 'res://drawable/default/CES-auryn-10ft-bg.png' : 'res://drawable/default/Gradient-Bottom-2.png'}>
        <Stack />
      </BackgroundImage>
    </Provider>
}
