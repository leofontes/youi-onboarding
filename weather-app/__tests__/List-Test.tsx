
import React from 'react';
import { List } from '../src/components';
import { fromApi } from '../src/adapters/dummyAdapter';

import renderer from 'react-test-renderer';
import { ListType } from '../src/components/list';

const assets = Array.from(Array(10)).map(() => fromApi(false));

test('renders correctly', () => {
  const tree = renderer.create(<List name="Discover" type={ListType.Featured} data={assets} />).toJSON();
  expect(tree).toMatchSnapshot();
});
