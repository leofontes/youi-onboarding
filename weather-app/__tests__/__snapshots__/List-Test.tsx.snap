// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <ListRef
    DEPRECATED_sendUpdatedChildFrames={false}
    alwaysBounceHorizontal={true}
    alwaysBounceVertical={false}
    data={
      Array [
        Array [
          Object {
            "details": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "episodes": 22,
            "extra": "Director: Alan Smithee | Starring: John Doe, Jane Doe",
            "genres": Array [
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 12,
                "name": "Adventure",
              },
            ],
            "id": 1234567,
            "images": Object {
              "Backdrop": "https://via.placeholder.com/1068x600",
              "Poster": "https://via.placeholder.com/200x300",
            },
            "key": "keystring",
            "releaseDate": "05/24/2015",
            "runtime": 3600,
            "seasons": 4,
            "similar": Array [],
            "thumbs": Object {
              "Backdrop": "https://via.placeholder.com/534x300",
              "Poster": "https://via.placeholder.com/400x600",
            },
            "title": "You.i TV",
            "type": "tv",
            "youtubeId": "nO_DIwuGBnA",
          },
          Object {
            "details": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "episodes": 22,
            "extra": "Director: Alan Smithee | Starring: John Doe, Jane Doe",
            "genres": Array [
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 12,
                "name": "Adventure",
              },
            ],
            "id": 1234567,
            "images": Object {
              "Backdrop": "https://via.placeholder.com/1068x600",
              "Poster": "https://via.placeholder.com/200x300",
            },
            "key": "keystring",
            "releaseDate": "05/24/2015",
            "runtime": 3600,
            "seasons": 4,
            "similar": Array [],
            "thumbs": Object {
              "Backdrop": "https://via.placeholder.com/534x300",
              "Poster": "https://via.placeholder.com/400x600",
            },
            "title": "You.i TV",
            "type": "tv",
            "youtubeId": "nO_DIwuGBnA",
          },
          Object {
            "details": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "episodes": 22,
            "extra": "Director: Alan Smithee | Starring: John Doe, Jane Doe",
            "genres": Array [
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 12,
                "name": "Adventure",
              },
            ],
            "id": 1234567,
            "images": Object {
              "Backdrop": "https://via.placeholder.com/1068x600",
              "Poster": "https://via.placeholder.com/200x300",
            },
            "key": "keystring",
            "releaseDate": "05/24/2015",
            "runtime": 3600,
            "seasons": 4,
            "similar": Array [],
            "thumbs": Object {
              "Backdrop": "https://via.placeholder.com/534x300",
              "Poster": "https://via.placeholder.com/400x600",
            },
            "title": "You.i TV",
            "type": "tv",
            "youtubeId": "nO_DIwuGBnA",
          },
        ],
        Array [
          Object {
            "details": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "episodes": 22,
            "extra": "Director: Alan Smithee | Starring: John Doe, Jane Doe",
            "genres": Array [
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 12,
                "name": "Adventure",
              },
            ],
            "id": 1234567,
            "images": Object {
              "Backdrop": "https://via.placeholder.com/1068x600",
              "Poster": "https://via.placeholder.com/200x300",
            },
            "key": "keystring",
            "releaseDate": "05/24/2015",
            "runtime": 3600,
            "seasons": 4,
            "similar": Array [],
            "thumbs": Object {
              "Backdrop": "https://via.placeholder.com/534x300",
              "Poster": "https://via.placeholder.com/400x600",
            },
            "title": "You.i TV",
            "type": "tv",
            "youtubeId": "nO_DIwuGBnA",
          },
          Object {
            "details": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "episodes": 22,
            "extra": "Director: Alan Smithee | Starring: John Doe, Jane Doe",
            "genres": Array [
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 12,
                "name": "Adventure",
              },
            ],
            "id": 1234567,
            "images": Object {
              "Backdrop": "https://via.placeholder.com/1068x600",
              "Poster": "https://via.placeholder.com/200x300",
            },
            "key": "keystring",
            "releaseDate": "05/24/2015",
            "runtime": 3600,
            "seasons": 4,
            "similar": Array [],
            "thumbs": Object {
              "Backdrop": "https://via.placeholder.com/534x300",
              "Poster": "https://via.placeholder.com/400x600",
            },
            "title": "You.i TV",
            "type": "tv",
            "youtubeId": "nO_DIwuGBnA",
          },
          Object {
            "details": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "episodes": 22,
            "extra": "Director: Alan Smithee | Starring: John Doe, Jane Doe",
            "genres": Array [
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 12,
                "name": "Adventure",
              },
            ],
            "id": 1234567,
            "images": Object {
              "Backdrop": "https://via.placeholder.com/1068x600",
              "Poster": "https://via.placeholder.com/200x300",
            },
            "key": "keystring",
            "releaseDate": "05/24/2015",
            "runtime": 3600,
            "seasons": 4,
            "similar": Array [],
            "thumbs": Object {
              "Backdrop": "https://via.placeholder.com/534x300",
              "Poster": "https://via.placeholder.com/400x600",
            },
            "title": "You.i TV",
            "type": "tv",
            "youtubeId": "nO_DIwuGBnA",
          },
        ],
        Array [
          Object {
            "details": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "episodes": 22,
            "extra": "Director: Alan Smithee | Starring: John Doe, Jane Doe",
            "genres": Array [
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 12,
                "name": "Adventure",
              },
            ],
            "id": 1234567,
            "images": Object {
              "Backdrop": "https://via.placeholder.com/1068x600",
              "Poster": "https://via.placeholder.com/200x300",
            },
            "key": "keystring",
            "releaseDate": "05/24/2015",
            "runtime": 3600,
            "seasons": 4,
            "similar": Array [],
            "thumbs": Object {
              "Backdrop": "https://via.placeholder.com/534x300",
              "Poster": "https://via.placeholder.com/400x600",
            },
            "title": "You.i TV",
            "type": "tv",
            "youtubeId": "nO_DIwuGBnA",
          },
          Object {
            "details": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "episodes": 22,
            "extra": "Director: Alan Smithee | Starring: John Doe, Jane Doe",
            "genres": Array [
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 12,
                "name": "Adventure",
              },
            ],
            "id": 1234567,
            "images": Object {
              "Backdrop": "https://via.placeholder.com/1068x600",
              "Poster": "https://via.placeholder.com/200x300",
            },
            "key": "keystring",
            "releaseDate": "05/24/2015",
            "runtime": 3600,
            "seasons": 4,
            "similar": Array [],
            "thumbs": Object {
              "Backdrop": "https://via.placeholder.com/534x300",
              "Poster": "https://via.placeholder.com/400x600",
            },
            "title": "You.i TV",
            "type": "tv",
            "youtubeId": "nO_DIwuGBnA",
          },
          Object {
            "details": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "episodes": 22,
            "extra": "Director: Alan Smithee | Starring: John Doe, Jane Doe",
            "genres": Array [
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 12,
                "name": "Adventure",
              },
            ],
            "id": 1234567,
            "images": Object {
              "Backdrop": "https://via.placeholder.com/1068x600",
              "Poster": "https://via.placeholder.com/200x300",
            },
            "key": "keystring",
            "releaseDate": "05/24/2015",
            "runtime": 3600,
            "seasons": 4,
            "similar": Array [],
            "thumbs": Object {
              "Backdrop": "https://via.placeholder.com/534x300",
              "Poster": "https://via.placeholder.com/400x600",
            },
            "title": "You.i TV",
            "type": "tv",
            "youtubeId": "nO_DIwuGBnA",
          },
        ],
        Array [
          Object {
            "details": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
            "episodes": 22,
            "extra": "Director: Alan Smithee | Starring: John Doe, Jane Doe",
            "genres": Array [
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 12,
                "name": "Adventure",
              },
            ],
            "id": 1234567,
            "images": Object {
              "Backdrop": "https://via.placeholder.com/1068x600",
              "Poster": "https://via.placeholder.com/200x300",
            },
            "key": "keystring",
            "releaseDate": "05/24/2015",
            "runtime": 3600,
            "seasons": 4,
            "similar": Array [],
            "thumbs": Object {
              "Backdrop": "https://via.placeholder.com/534x300",
              "Poster": "https://via.placeholder.com/400x600",
            },
            "title": "You.i TV",
            "type": "tv",
            "youtubeId": "nO_DIwuGBnA",
          },
        ],
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    getItemLayout={[Function]}
    horizontal={true}
    initialNumToRender={2}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    name="Discover"
    numColumns={1}
    onContentSizeChange={null}
    onEndReachedThreshold={2}
    onLayout={[MockFunction]}
    onMomentumScrollBegin={[MockFunction]}
    onMomentumScrollEnd={[MockFunction]}
    onResponderGrant={[MockFunction]}
    onResponderReject={[MockFunction]}
    onResponderRelease={[MockFunction]}
    onResponderTerminationRequest={[MockFunction]}
    onScroll={[MockFunction]}
    onScrollBeginDrag={[MockFunction]}
    onScrollEndDrag={[MockFunction]}
    onScrollShouldSetResponder={[MockFunction]}
    onStartShouldSetResponder={[MockFunction]}
    onStartShouldSetResponderCapture={[MockFunction]}
    onTouchCancel={[MockFunction]}
    onTouchEnd={[MockFunction]}
    onTouchMove={[MockFunction]}
    onTouchStart={[MockFunction]}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollBarThumbImage={null}
    scrollEnabled={true}
    scrollEventThrottle={50}
    sendMomentumEvents={true}
    snapToEnd={true}
    snapToStart={true}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "overflow": "scroll",
        },
        undefined,
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <ScrollContentData
      collapsable={false}
      onLayout={[MockFunction]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <View
            style={null}
          >
            <Composition
              onTimelinesLoaded={[Function]}
              source="Auryn_Container-Btn-Backdrop-Large"
            >
              <YiPushButtonView
                name="Btn-Backdrop-Large"
                onCompositionDidLoad={[Function]}
                onFocus={[Function]}
                onPress={[Function]}
                onTimelinesLoaded={[Function]}
              >
                <YiImageView
                  name="Image-Dynamic"
                  onTimelinesLoaded={[Function]}
                  source={
                    Object {
                      "uri": "https://via.placeholder.com/1068x600",
                    }
                  }
                />
                <YiTextSceneNode
                  name="Text-Title"
                  style={
                    Object {
                      "color": "#f6f1ee",
                    }
                  }
                  text="You.i TV"
                />
                <YiTextSceneNode
                  name="Text-Metadata"
                  text="tv"
                />
                <YiTextSceneNode
                  name="Unfocus-Text-Title"
                  text="You.i TV"
                />
                <YiTextSceneNode
                  name="Text-Details"
                  text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
                />
              </YiPushButtonView>
            </Composition>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <View
              style={null}
            >
              <Composition
                onTimelinesLoaded={[Function]}
                source="Auryn_Container-Btn-Backdrop-Small"
              >
                <YiPushButtonView
                  name="Btn-Backdrop-Small"
                  onCompositionDidLoad={[Function]}
                  onFocus={[Function]}
                  onPress={[Function]}
                  onTimelinesLoaded={[Function]}
                >
                  <YiImageView
                    name="Image-Dynamic"
                    onTimelinesLoaded={[Function]}
                    source={
                      Object {
                        "uri": "https://via.placeholder.com/534x300",
                      }
                    }
                  />
                  <YiTextSceneNode
                    name="Text-Title"
                    style={
                      Object {
                        "color": "#f6f1ee",
                      }
                    }
                    text="You.i TV"
                  />
                  <YiTextSceneNode
                    name="Text-Metadata"
                    text="tv"
                  />
                  <YiTextSceneNode
                    name="Unfocus-Text-Title"
                    text="You.i TV"
                  />
                </YiPushButtonView>
              </Composition>
            </View>
            <View
              style={null}
            >
              <Composition
                onTimelinesLoaded={[Function]}
                source="Auryn_Container-Btn-Backdrop-Small"
              >
                <YiPushButtonView
                  name="Btn-Backdrop-Small"
                  onCompositionDidLoad={[Function]}
                  onFocus={[Function]}
                  onPress={[Function]}
                  onTimelinesLoaded={[Function]}
                >
                  <YiImageView
                    name="Image-Dynamic"
                    onTimelinesLoaded={[Function]}
                    source={
                      Object {
                        "uri": "https://via.placeholder.com/534x300",
                      }
                    }
                  />
                  <YiTextSceneNode
                    name="Text-Title"
                    style={
                      Object {
                        "color": "#f6f1ee",
                      }
                    }
                    text="You.i TV"
                  />
                  <YiTextSceneNode
                    name="Text-Metadata"
                    text="tv"
                  />
                  <YiTextSceneNode
                    name="Unfocus-Text-Title"
                    text="You.i TV"
                  />
                </YiPushButtonView>
              </Composition>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <View
              style={null}
            >
              <Composition
                onTimelinesLoaded={[Function]}
                source="Auryn_Container-Btn-Backdrop-Small"
              >
                <YiPushButtonView
                  name="Btn-Backdrop-Small"
                  onCompositionDidLoad={[Function]}
                  onFocus={[Function]}
                  onPress={[Function]}
                  onTimelinesLoaded={[Function]}
                >
                  <YiImageView
                    name="Image-Dynamic"
                    onTimelinesLoaded={[Function]}
                    source={
                      Object {
                        "uri": "https://via.placeholder.com/534x300",
                      }
                    }
                  />
                  <YiTextSceneNode
                    name="Text-Title"
                    style={
                      Object {
                        "color": "#f6f1ee",
                      }
                    }
                    text="You.i TV"
                  />
                  <YiTextSceneNode
                    name="Text-Metadata"
                    text="tv"
                  />
                  <YiTextSceneNode
                    name="Unfocus-Text-Title"
                    text="You.i TV"
                  />
                </YiPushButtonView>
              </Composition>
            </View>
            <View
              style={null}
            >
              <Composition
                onTimelinesLoaded={[Function]}
                source="Auryn_Container-Btn-Backdrop-Small"
              >
                <YiPushButtonView
                  name="Btn-Backdrop-Small"
                  onCompositionDidLoad={[Function]}
                  onFocus={[Function]}
                  onPress={[Function]}
                  onTimelinesLoaded={[Function]}
                >
                  <YiImageView
                    name="Image-Dynamic"
                    onTimelinesLoaded={[Function]}
                    source={
                      Object {
                        "uri": "https://via.placeholder.com/534x300",
                      }
                    }
                  />
                  <YiTextSceneNode
                    name="Text-Title"
                    style={
                      Object {
                        "color": "#f6f1ee",
                      }
                    }
                    text="You.i TV"
                  />
                  <YiTextSceneNode
                    name="Text-Metadata"
                    text="tv"
                  />
                  <YiTextSceneNode
                    name="Unfocus-Text-Title"
                    text="You.i TV"
                  />
                </YiPushButtonView>
              </Composition>
            </View>
          </View>
          <View
            style={null}
          >
            <Composition
              onTimelinesLoaded={[Function]}
              source="Auryn_Container-Btn-Backdrop-Large"
            >
              <YiPushButtonView
                name="Btn-Backdrop-Large"
                onCompositionDidLoad={[Function]}
                onFocus={[Function]}
                onPress={[Function]}
                onTimelinesLoaded={[Function]}
              >
                <YiImageView
                  name="Image-Dynamic"
                  onTimelinesLoaded={[Function]}
                  source={
                    Object {
                      "uri": "https://via.placeholder.com/1068x600",
                    }
                  }
                />
                <YiTextSceneNode
                  name="Text-Title"
                  style={
                    Object {
                      "color": "#f6f1ee",
                    }
                  }
                  text="You.i TV"
                />
                <YiTextSceneNode
                  name="Text-Metadata"
                  text="tv"
                />
                <YiTextSceneNode
                  name="Unfocus-Text-Title"
                  text="You.i TV"
                />
                <YiTextSceneNode
                  name="Text-Details"
                  text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
                />
              </YiPushButtonView>
            </Composition>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "width": 2136,
          }
        }
      />
    </ScrollContentData>
  </ListRef>,
  <ViewRef
    name="Discover"
    onCompositionDidLoad={[Function]}
    onTimelinesLoaded={[Function]}
  />,
]
`;
